conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,571 @@
+Labels: check, commit-action, error, optional-type, transaction-stmt, transactional-expr
+
+function init() returns error? {
+    io:println(transactional); // output false
+    transaction {
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+    io:println(transactional); // output false
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, function-call-expr, optional-type, transaction-stmt, transactional-expr,
+        transactional-function
+
+function init() returns error? {
+    transaction {
+        f1(); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+transactional function f1() {
+    io:println(transactional);
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, function-call-expr, optional-type, transaction-stmt, transactional-expr,
+        transactional-function
+
+function init() returns error? {
+    transactional function () trxFunc = transactional function() {
+        io:println(transactional);
+    };
+
+    transaction {
+        trxFunc(); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, explicit-anonymous-function-expr, isolated-function, optional-type,
+        transaction-stmt, transaction:Info, transaction:onCommit, transactional-expr, var
+
+function init() returns error? {
+    var onCommitFunc = isolated function(transaction:Info? info) {
+        io:println(transactional);
+    };
+
+    transaction {
+        transaction:onCommit(onCommitFunc); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, explicit-anonymous-function-expr, if-else-stmt, int, is-expr, isolated-function,
+        optional-type, rollback-stmt, transaction-stmt, transaction:Info, transaction:onRollback, transactional-expr,
+        var
+
+function init() returns error? {
+    var onRollbackFunc = isolated function(transaction:Info? info, error? cause, boolean willTry) {
+        io:println(transactional);
+    };
+
+    transaction {
+        int? a = 10;
+
+        if a is int {
+            io:println(transactional); // output true
+            transaction:onRollback(onRollbackFunc); // output true
```

Comment:
Same as in https://github.com/ballerina-platform/ballerina-spec/pull/1054/files#r832467849